File: src/main/java/no/nav/sosialhjelp/soknad/web/rest/providers/SoknadObjectMapperProvider.java
package no.nav.sosialhjelp.soknad.web.rest.providers;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import com.fasterxml.jackson.module.kotlin.KotlinModule;

import javax.ws.rs.ext.ContextResolver;
import javax.ws.rs.ext.Provider;

@Provider
public class SoknadObjectMapperProvider implements ContextResolver<ObjectMapper> {

    private ObjectMapper objectMapper;

    public SoknadObjectMapperProvider() {
        objectMapper = createObjectMapper();
    }

    @Override
    public ObjectMapper getContext(Class<?> type) {
        return objectMapper;
    }

    private static ObjectMapper createObjectMapper() {
        final ObjectMapper mapper = new ObjectMapper();
        mapper.registerModule(new JavaTimeModule());
        mapper.registerModule(new KotlinModule());
        mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
        return mapper;
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/app/rest/provider/SoknadObjectMapperProvider.kt
// package no.nav.sosialhjelp.soknad.app.rest.provider
//
// import com.fasterxml.jackson.databind.ObjectMapper
// import com.fasterxml.jackson.databind.SerializationFeature
// import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
// import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
// import com.fasterxml.jackson.module.kotlin.registerKotlinModule
// import javax.ws.rs.ext.ContextResolver
// import javax.ws.rs.ext.Provider
//
// @Provider
// class SoknadObjectMapperProvider : ContextResolver<ObjectMapper> {
//
//    private val objectMapper: ObjectMapper = createObjectMapper()
//
//    override fun getContext(type: Class<*>?): ObjectMapper {
//        return objectMapper
//    }
//
//    companion object {
//        private fun createObjectMapper(): ObjectMapper {
//            return jacksonObjectMapper()
//                .registerKotlinModule()
//                .registerModule(JavaTimeModule())
//                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
//        }
//    }
// }
